src/components/Radio/Radio.test.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { render } from 'test';
+import Radio from './Radio';
+import { fireEvent } from '../../test';
+
+describe('Radio', () => {
+  it('should render correctly', () => {
+    const { container } = render(<Radio />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render correctly with color', () => {
+    const { container } = render(<Radio color={'teal-400'} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should change to checked on click', () => {
+    const mockOnClick = jest.fn();
+    const { getByTestId } = render(<Radio onChange={mockOnClick} dataTestId={'test-check'} />);
+
+    const radio = getByTestId('radio-input-test-check');
+
+    fireEvent.click(radio.children[0]);
+
+    expect(mockOnClick).toHaveBeenCalledTimes(1);
+    expect((radio.children[0] as HTMLInputElement).checked).toBeTruthy();
+  });
+
+  it('disabled should not change to checked on click', () => {
+    const mockOnClick = jest.fn();
+    const { getByTestId } = render(
+      <Radio disabled={true} onChange={mockOnClick} dataTestId={'test-disabled'} />
+    );
+
+    const radio = getByTestId('radio-input-test-disabled');
+
+    fireEvent.click(radio.children[0]);
+
+    expect(mockOnClick).toHaveBeenCalledTimes(0);
+    expect((radio.children[0] as HTMLInputElement).checked).toBeFalsy();
```

Comment:
I would suggest using a different approach here as picking the first element may not work in the future (e.g. we add a new span element before the input element). 

example:
```
   const { container } = render(
      <Radio disabled={true} onChange={mockOnClick} dataTestId={'test-disabled'} />
    );

    const radio = container.querySelector('input[type="radio"]') as HTMLInputElement;

    fireEvent.click(radio);

    expect(radio.checked).toBeFalsy();
```

this ensures that we always pick and test the input element.